flytekit/core/map_task.py

```diff
@@ -1,72 +1,168 @@
+        from these individual outputs as the final output value.
+        """
+        ctx = FlyteContext.current_context()
+        if ctx.execution_state is not None and ctx.execution_state.mode == ExecutionState.Mode.LOCAL_WORKFLOW_EXECUTION:
+            # In workflow execution mode we actually need to use the parent (mapper) task output interface.
+            return self._python_interface.outputs[k]
+        return self._run_task._python_interface.outputs[k]
+
+    def _execute_map_task(self, ctx: FlyteContext, **kwargs) -> Any:
+        """
+        This is called during ExecutionState.Mode.TASK_EXECUTION executions, that is executions orchestrated by the
+        Flyte platform. Individual instances of the map task, aka array task jobs are passed the full set of inputs but
+        only produce a single output based on the map task (array task) instance. The array plugin handler will actually
+        create a collection from these individual outputs as the final map task output value.
+        """
+        task_index = self._compute_array_job_index()
+        map_task_inputs = {}
+        for k in self.interface.inputs.keys():
+            map_task_inputs[k] = kwargs[k][task_index]
+        return self._run_task.execute(**map_task_inputs)
+
+    def _raw_execute(self, **kwargs) -> Any:
+        """
+        This is called during locally run executions. Unlike array task execution on the Flyte platform, _raw_execute
+        produces the full output collection.
+        """
         outputs_expected = True
         if not self.interface.outputs:
             outputs_expected = False
         outputs = []
-        for k in self.interface.outputs.keys():
+        for _ in self._outputs_interface.keys():
             outputs.append([])
-        for i in range(len(kwargs[any_key])):
+
+        any_input_key = (
+            list(self._run_task.interface.inputs.keys())[0]
+            if self._run_task.interface.inputs.items() is not None
+            else None
+        )
+
+        for i in range(len(kwargs[any_input_key])):
+            single_instance_inputs = {}
             for k in self.interface.inputs.keys():
-                batch[k] = kwargs[k][i]
-            o = self._run_task.execute(**batch)
+                single_instance_inputs[k] = kwargs[k][i]
+            o = self._run_task.execute(**single_instance_inputs)
             if outputs_expected:
                 for x in range(len(outputs)):
                     outputs[x].append(o[x])
+
         if len(outputs) == 1:
             return outputs[0]
 
         return tuple(outputs)
 
 
-def maptask(tk: PythonTask, concurrency="auto", metadata=None):
-    if not isinstance(tk, PythonTask):
-        raise ValueError(f"Only Flyte Task types are supported in maptask currently, received {type(tk)}")
+def maptask(tk: PythonFunctionTask, concurrency=None, min_success_ratio=None, metadata=None):
```

Comment:
lets have a usage example here

Reply:
what does concurrency=None mean? means full cardinality?
what is concurrency != length of the array?

Reply:
done